src/frontend/css/plugins/color.js

```diff
@@ -0,0 +1,69 @@
+const colorFunctions = require('less/lib/less/functions/color').default;
+
+const SUPPORTED_COLOR_FUNCTIONS = {
+  LIGHTEN: 'lighten',
+  DARKEN: 'darken',
+  FADE: 'fade'
+};
+
+// TODO (esn-frontend-common-libs#51): Write tests for this plugin
+function colorFunctionInterceptor(color, amount, method, functionName) {
+  if (!Object.values(SUPPORTED_COLOR_FUNCTIONS).includes(functionName)) {
+    throw new Error(`The LESS's color function '${functionName}' is not supported with CSS Variables`);
+  }
+
+  // If the "color" argument is a known, concrete color, not a CSS Variable, we'll just use the default LESS function.
+  if (color.toHSL) return colorFunctions[functionName](color, amount, method);
+
+  // If the "color" argument is not a transformed one, we'll first transform it here.
+  if (color.args) return _firstTransform();
+
+  // Else, if the current "color" argument has already been transformed at least once, its value will be just a string.
+  // e.g. hsl(var(--primary-color-h), var(--primary-color-s), calc(var(--primary-color-l) + 20%)). Therefore we need to process it as a string.
+  return _nthTransform();
+
+  function _firstTransform() {
```

Comment:
Technically, we are transforming color value in both cases, so there's no _transformColorArgs because the color args are not transformed. The only difference is whether we're transforming it for the first time (color.args is there) or not (color.args is not there). color.args is not what we're transforming but what we use to transform.